tools/memory-model-test/memory_model_test.cpp

```diff
@@ -13,13 +13,22 @@
 
 #include <iostream>
 #include <list>
-#include <semaphore>
 #include <thread>
 #include <chrono>
 #include <random>
 #include <utility>
 #include <unistd.h>
 
+#ifdef AVOID_CPP20_SEMAPHORE
```

Comment:
Is it possible/desirable to replace this with an automated test? From the docs it would seem that `#if __cpp_lib_semaphore >= 201907L` (or something along those lines) would do the trick.

Reply:
That's worth investigating, but will take some testing with different compilers to confirm.

Reply:
No problem, it's a nice-to-have we can fix this some other time - the current code looks to work fine.